lnwallet/channel.go

```diff
@@ -3180,8 +3180,20 @@ func (lc *LightningChannel) validateCommitmentSanity(theirLogCounter,
 	ourInitialBalance := commitChain.tip().ourBalance
 	theirInitialBalance := commitChain.tip().theirBalance
 
-	ourBalance, theirBalance, commitWeight, filteredView, feePerKw :=
-		lc.computeView(view, remoteChain, false)
+	ourBalance, theirBalance, commitWeight, filteredView, feePerKw := lc.computeView(
+		view, remoteChain, false,
+	)
+
+	// As a quick sanity check, we'll ensure that if we interpret the
+	// balances as signed integers, they haven't dipped down below zero. If
+	// they have, then this indicates that a party doesn't have sufficient
+	// balance to satisfy the final evaluated HTLC's.
+	switch {
+	case int64(ourBalance) < 0:
```

Comment:
It took me a while to understand this check. I think it would make more sense to check in `processAddEntry` that the balance would not go below 0 *before* subtracting (though that would require adding an error return value to that function). There may be pathological cases here as well, where a bunch of high value adds are all applied and underflow the balance so much that the high order bit gets unset.

Reply:
`computeView` doesn't currently return an error. Seems easier just to verify at the very end as it's less over all code disruption. 

Agree about the pathological case...if this happens though, it should fail the `CheckTransactionSanity` check at the very end.